jwt-cpp/0.7.0 : "picojson.h" is not found

### Description

jwt-cpp/include/jwt-cpp/traits/kazuho-picojson/traits.h:7:10: fatal error: picojson.h: No such file or directory #include "picojson.h"

There is no "picojson.h" in the package; and the picojson library is not a dependency. It is not clear which "picojson.h" should be used and why.

### Package and Environment Details

* Package Name/Version: **jwt-cpp/0.7.0**
* Operating System+version: **Linux rhel7**
* Compiler+version: **GCC 8**
* Conan version: **conan 1.65.0**
* Python version: **Python 3.10.10**


### Conan profile

[settings]
os=Linux
os_build=Linux
os.distro=ubi7
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8.2
compiler.libcxx=libstdc++11
compiler.cppstd=17
build_type=Release



### Steps to reproduce

conan install --build=missing .

<conanfile.txt>
[requires]
jwt-cpp/0.7.0
openssl/1.1.1s
zlib/1.2.13
[options]
*:shared=True

### Logs

<details><summary>Click to expand log</summary>

```
jwt-cpp/include/jwt-cpp/traits/kazuho-picojson/traits.h:7:10: fatal error: picojson.h: No such file or directory #include "picojson.h"

```

</details>

--- a/recipes/jwt-cpp/all/conanfile.py
+++ b/recipes/jwt-cpp/all/conanfile.py
@@ -16,9 +16,12 @@ class JwtCppConan(ConanFile):
     package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
 
+    options = { "enable_picojson": [True, False] }
+    default_options = { "enable_picojson" : False }
+
     @property
-    def _supports_generic_json(self):
-        return Version(self.version) >= "0.5.0"
+    def _enable_picojson(self):
+        return Version(self.version) < "0.5.0" or self.options.enable_picojson
 
     def export_sources(self):
         export_conandata_patches(self)
@@ -28,7 +31,7 @@ class JwtCppConan(ConanFile):
 
     def requirements(self):
         self.requires("openssl/[>=1.1 <4]")
-        if not self._supports_generic_json:
+        if self._enable_picojson:
             self.requires("picojson/1.3.0")
 
     def package_id(self):
@@ -48,9 +51,11 @@ class JwtCppConan(ConanFile):
     def package_info(self):
         self.cpp_info.bindirs = []
         self.cpp_info.libdirs = []
-
+        self.cpp_info.requires = ["openssl::openssl"]
+        if self._enable_picojson:
+            self.cpp_info.requires.append("picojson::picojson")
+        else:
+            self.cpp_info.defines.append("JWT_DISABLE_PICOJSON")
         self.cpp_info.set_property("cmake_file_name", "jwt-cpp")
         self.cpp_info.set_property("cmake_target_name", "jwt-cpp::jwt-cpp")
 
-        if self._supports_generic_json:
-            self.cpp_info.defines.append("JWT_DISABLE_PICOJSON")

--- a/recipes/jwt-cpp/all/conanfile.py
+++ b/recipes/jwt-cpp/all/conanfile.py
@@ -16,12 +16,12 @@ class JwtCppConan(ConanFile):
     package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
 
-    options = { "enable_picojson": [True, False] }
-    default_options = { "enable_picojson" : False }
+    options = { "with_picojson": [True, False] }
+    default_options = { "with_picojson" : False }
 
     @property
-    def _enable_picojson(self):
-        return Version(self.version) < "0.5.0" or self.options.enable_picojson
+    def _with_picojson(self):
+        return Version(self.version) < "0.5.0" or self.options.with_picojson
 
     def export_sources(self):
         export_conandata_patches(self)
@@ -31,7 +31,7 @@ class JwtCppConan(ConanFile):
 
     def requirements(self):
         self.requires("openssl/[>=1.1 <4]")
-        if self._enable_picojson:
+        if self._with_picojson:
             self.requires("picojson/1.3.0")
 
     def package_id(self):
@@ -52,7 +52,7 @@ class JwtCppConan(ConanFile):
         self.cpp_info.bindirs = []
         self.cpp_info.libdirs = []
         self.cpp_info.requires = ["openssl::openssl"]
-        if self._enable_picojson:
+        if self._with_picojson:
             self.cpp_info.requires.append("picojson::picojson")
         else:
             self.cpp_info.defines.append("JWT_DISABLE_PICOJSON")

